### PR TITLE
Company configuration improvement in updateUser() method

### PR DESCRIPTION
**Pull request by mozhaiskyi**
It contains implementations for Android, iOS, and WEB.

## API changed
- added the `Company` model which represents Intercome company configuration
- removed `company` and `companyId` fields in `updateUser()` method

### CODE REVIEW

File: intercom_flutter/test/intercom_flutter_test.dart [hunk 1]
```diff
@@ -157,13 +157,17 @@ void main() {
     });
 
     test('updateUser', () {
+      final company = Company(
+        id: 'some_company_llc',
+        name: 'Soma Company LLC',
```
Comment (Zazo032) [inline intercom_flutter/test/intercom_flutter_test.dart:167]:
```suggestion
        name: 'Some Company LLC',
```

File: intercom_flutter/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt [hunk 2]
```diff
@@ -277,10 +276,34 @@ class IntercomFlutterPlugin : FlutterPlugin, MethodCallHandler, EventChannel.Str
...
     }
 
-    if (company != null && companyId != null) {
+    if (company != null) {
       val icmCompany = Company.Builder()
-      icmCompany.withName(company)
-      icmCompany.withCompanyId(companyId)
+
+      if (company.containsKey("id")) {
+        icmCompany.withCompanyId(company["id"] as String?)
```
Comment (Zazo032) [inline intercom_flutter/android/src/main/kotlin/io/maido/intercom/IntercomFlutterPlugin.kt:286]: According to the Dart code, id should be non-null, why is this one nullable?

File: intercom_flutter_platform_interface/lib/intercom_flutter_platform_interface.dart [hunk 3]
```diff
@@ -118,8 +118,7 @@ abstract class IntercomFlutterPlatform extends PlatformInterface {
     String? email,
     String? name,
     String? phone,
-    String? company,
-    String? companyId,
+    Map<String, dynamic>? company,
```
Comment (Zazo032) [inline intercom_flutter_platform_interface/lib/intercom_flutter_platform_interface.dart:121]: Why is this not using the `Company` object?

**Review by Zazo032 — COMMENTED** [review]
Just a few small comments, make sure to update the `pubspec` versions to a new one so this can be published

**Comment by eduardo-pellenz** [comment]
hey @Zazo032, is there a reason why this wasn't merged?
We could really use it right now

If this PR need changes, I could do that.

**Comment by Zazo032** [comment]
> hey @Zazo032, is there a reason why this wasn't merged?
> 
> We could really use it right now
> 
> 
> 
> If this PR need changes, I could do that.

I left some comments 2 years ago but haven't heard back, not sure if the PR even works with the current API or would require some adaptations.

@mozhaiskyi could you have a look?
If not, @eduardo-pellenz feel free to cherry pick the necessary changes, check if it needs any extra changes and open a new PR, I can have a look at it

**Comment by mariopepe** [comment]
@Zazo032 Hi Cristian, :D i have created a new PR and addrssed the 3 comments you left here, please let me know if anything else needs to be done thanks!

this is the new PR #460 

cc @eduardo-pellenz @mozhaiskyi

**Comment by Zazo032** [comment]
Thanks @mariopepe - let's follow up in the new PR

**Pull request closed without merging**